go/vt/sqlparser/utils.go

```diff
@@ -135,15 +136,31 @@ func ReplaceTableQualifiers(query, olddb, newdb string) (string, error) {
 	upd := Rewrite(in, func(cursor *Cursor) bool {
 		switch node := cursor.Node().(type) {
 		case TableName:
-			if !node.Qualifier.IsEmpty() &&
-				node.Qualifier.String() == oldQualifier.String() {
+			if node.Qualifier.String() == oldQualifier.String() {
 				node.Qualifier = newQualifier
 				cursor.Replace(node)
 				modified = true
 			}
+		case *CreateDatabase:
+			if node.DBName.String() == oldQualifier.String() {
+				node.DBName = newQualifier
+				cursor.Replace(node)
+				modified = true
+			}
+		case *AlterDatabase:
+			if node.DBName.String() == oldQualifier.String() {
+				node.DBName = newQualifier
+				cursor.Replace(node)
+				modified = true
+			}
+		case *DropDatabase:
+			if node.DBName.String() == oldQualifier.String() {
+				node.DBName = newQualifier
+				cursor.Replace(node)
+				modified = true
+			}
```

Comment:
Any reason not to consolidate these?

Reply:
Yes - `golang` will give compilation errors, because `SQLNode` (the shared interface for all these types) does not have a `DBName` field (obviously as it is an interface, and each implementing type adds the fields independently).